diff --git a/tap-snapshots/test-arborist-reify.js-TAP.test.js b/tap-snapshots/test-arborist-reify.js-TAP.test.js
index a0d6d51..3a026d8 100644
--- a/tap-snapshots/test-arborist-reify.js-TAP.test.js
+++ b/tap-snapshots/test-arborist-reify.js-TAP.test.js
@@ -46018,10 +46018,51 @@ Object {
 
 exports[`test/arborist/reify.js TAP workspaces root as-a-workspace > should produce expected package-lock file 1`] = `
 Object {
+  "dependencies": Object {
+    "a": Object {
+      "requires": Object {
+        "abbrev": "^1.0.0",
+      },
+      "version": "file:a",
+    },
+    "abbrev": Object {
+      "integrity": "sha512-nne9/IiQ/hzIhY6pdDnbBtz7DjPTKrY00P/zvPSm5pOFkl6xuGrGnXn/VtTNNfNtAfZ9/1RtehkszU9qcTii0Q==",
+      "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
+      "version": "1.1.1",
+    },
+    "b": Object {
+      "version": "file:b",
+    },
+    "workspaces-root-linked": Object {
+      "dependencies": Object {
+        "a": Object {
+          "requires": Object {
+            "abbrev": "^1.0.0",
+          },
+          "version": "file:a",
+        },
+        "abbrev": Object {
+          "integrity": "sha512-nne9/IiQ/hzIhY6pdDnbBtz7DjPTKrY00P/zvPSm5pOFkl6xuGrGnXn/VtTNNfNtAfZ9/1RtehkszU9qcTii0Q==",
+          "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
+          "version": "1.1.1",
+        },
+        "b": Object {
+          "version": "file:b",
+        },
+      },
+      "requires": Object {
+        "a": "file:{CWD}/test/arborist/reify-workspaces-root-as-a-workspace/a",
+        "b": "file:{CWD}/test/arborist/reify-workspaces-root-as-a-workspace/b",
+        "workspaces-root-linked": "file:{CWD}/test/arborist/reify-workspaces-root-as-a-workspace",
+      },
+      "version": "file:",
+    },
+  },
   "lockfileVersion": 2,
   "name": "workspaces-root-linked",
   "packages": Object {
     "": Object {
+      "name": "workspaces-root-linked",
       "version": "1.0.0",
       "workspaces": Array [
         ".",
@@ -46029,6 +46070,32 @@ Object {
         "b",
       ],
     },
+    "a": Object {
+      "dependencies": Object {
+        "abbrev": "^1.0.0",
+      },
+      "version": "1.0.0",
+    },
+    "b": Object {
+      "version": "1.0.0",
+    },
+    "node_modules/a": Object {
+      "link": true,
+      "resolved": "a",
+    },
+    "node_modules/abbrev": Object {
+      "integrity": "sha512-nne9/IiQ/hzIhY6pdDnbBtz7DjPTKrY00P/zvPSm5pOFkl6xuGrGnXn/VtTNNfNtAfZ9/1RtehkszU9qcTii0Q==",
+      "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
+      "version": "1.1.1",
+    },
+    "node_modules/b": Object {
+      "link": true,
+      "resolved": "b",
+    },
+    "node_modules/workspaces-root-linked": Object {
+      "link": true,
+      "resolved": "",
+    },
   },
   "requires": true,
   "version": "1.0.0",
